flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/PrintSinkFunctionTest.java

```diff
@@ -0,0 +1,234 @@
+
+/**
+ * Tests for the {@link org.apache.flink.streaming.api.functions.sink.PrintSinkFunction}.
+ */
+public class PrintSinkFunctionTest<IN> extends RichSinkFunction<IN> {
+
+	private Environment envForPrefixNull = new Environment() {
+		@Override
+		public JobID getJobID() {
+			return null;
+		}
+
+		@Override
+		public JobVertexID getJobVertexId() {
+			return null;
+		}
+
+		@Override
+		public ExecutionAttemptID getExecutionId() {
+			return null;
+		}
+
+		@Override
+		public Configuration getTaskConfiguration() {
+			return null;
+		}
+
+		@Override
+		public TaskManagerRuntimeInfo getTaskManagerInfo() {
+			return null;
+		}
+
+		@Override
+		public Configuration getJobConfiguration() {
+			return null;
+		}
+
+		@Override
+		public int getNumberOfSubtasks() {
+			return 0;
+		}
+
+		@Override
+		public int getIndexInSubtaskGroup() {
+			return 0;
+		}
+
+		@Override
+		public InputSplitProvider getInputSplitProvider() {
+			return null;
+		}
+
+		@Override
+		public IOManager getIOManager() {
+			return null;
+		}
+
+		@Override
+		public MemoryManager getMemoryManager() {
+			return null;
+		}
+
+		@Override
+		public String getTaskName() {
+			return null;
+		}
+
+		@Override
+		public String getTaskNameWithSubtasks() {
+			return null;
+		}
+
+		@Override
+		public ClassLoader getUserClassLoader() {
+			return null;
+		}
+
+		@Override
+		public Map<String, Future<Path>> getDistributedCacheEntries() {
+			return null;
+		}
+
+		@Override
+		public BroadcastVariableManager getBroadcastVariableManager() {
+			return null;
+		}
+
+		@Override
+		public AccumulatorRegistry getAccumulatorRegistry() {
+			return null;
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId) {
+
+		}
+
+		@Override
+		public void acknowledgeCheckpoint(long checkpointId, StateHandle<?> state) {
+
+		}
+
+		@Override
+		public ResultPartitionWriter getWriter(int index) {
+			return null;
+		}
+
+		@Override
+		public ResultPartitionWriter[] getAllWriters() {
+			return new ResultPartitionWriter[0];
+		}
+
+		@Override
+		public InputGate getInputGate(int index) {
+			return null;
+		}
+
+		@Override
+		public InputGate[] getAllInputGates() {
+			return new InputGate[0];
+		}
+	};
+
+	@Test
+	public void testPrintSinkStdOut(){
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		final HashMap<String, Accumulator<?, ?>> accumulators = new HashMap<String, Accumulator<?, ?>>();
+		final StreamingRuntimeContext ctx = new StreamingRuntimeContext(
+				"Test Print Sink",
+				this.envForPrefixNull,
+				null,
+				executionConfig,
+				null,
+				null,
+				accumulators
+		);
+
+		PrintSinkFunction<String> printSink = new PrintSinkFunction<String>();
+		printSink.setRuntimeContext(ctx);
+		try {
+			printSink.open(new Configuration());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		printSink.setTargetToStandardOut();
+		printSink.invoke("hello world!");
+
+		assertEquals("Print to System.out", printSink.toString());
```

Comment:
As I see, the sink actually prints is printSink.invoke("hello world!");.
And in the function invoke, it just calls stream.println(record.toString());.
Could you tell me how to get the prints from the stream?
Because I think assertEquals("hello world!", record); is just like assertEquals("hello world!", "hello world!");  and it`s not necessary.